packages/frontend/src/components/DataViz/visualizations/ChartView.tsx

```diff
@@ -75,11 +76,9 @@ const ChartView = memo(
         }
         const loading = isLoadingProp || transformLoading;
 
-        // TODO: this could be more robust
-        const errorMessage = error?.message?.includes('Binder Error')
-            ? 'Some specified columns do not exist in the data'
+        const errorMessage = isApiError(error)
```

Comment:
This error type checking should be deeper. At least in useChart or maybe deeper but that might need type changes. The `Error` type comes from `useAsync`

Reply:
done here! https://github.com/lightdash/lightdash/pull/11324/commits/3d886839911049d468e5c50ad846d12775abd639